articles/openshift/oshift-how-use-portworx-shared-volumes.md

```diff
@@ -0,0 +1,238 @@
+services: openshift
+author: Ben Bacon
+reviewer:
+lastreviewed: 16/10/2019
+
+toc_rootlink: How To
+toc_sub1: Use Portworx with OpenShift
+toc_sub2:
+toc_sub3:
+toc_sub4:
+toc_title: Use Portworx shared volumes
+toc_fullpath: How To/Use Portworx with OpenShift/oshift-how-use-portworx-shared-volumes.md
+toc_mdlink: oshift-how-use-portworx-shared-volumes.md
+---
+
+# How to use Portworx shared volumes
+
+## Overview
+
+Portworx is a cloud-native storage solution that is now available as an integration with our OpenShift offering. This article describes how to create shared volumes.
+
+### Intended audience
+
+This article assumes you have access to a Portworx integrated OpenShift 3.11 cluster and that you have cluster-admin rights. It also assumes familiarity with `oc`, the OpenShift command-line client. 
+
+If you're interested in a free 30 day trial of Portworx, raise a Service Request via the [My Calls](https://portal.skyscapecloud.com/support/ivanti) section of the UKCloud Portal.
+
+## Shared volumes
+
+Portworx enables persistent volumes claims to be mounted to multiple containers across multiple nodes enabling ease of pod-scaling.
+
+## Storage classes
+
+We create a number of storage classes when provisioning an OpenShift cluster with the Portworx integration. The following example shows a storage class from which shared persistent volume claims (pvc) with a single replica can be provisioned:
+
+```none
+kind: StorageClass
+apiVersion: storage.k8s.io/v1beta1
+metadata:
+    name: portworx-repl1-shared
+provisioner: com.openstorage.pxd
+parameters:
+   repl: "1"
+   shared: "true"
+```
+
+If you want to use different attributes to the pre-provisioned storage classes, include the following values within the spec:
+
+```none
+provisioner: com.openstorage.pxd
+parameters:
+	shared: "true"
+```
+
+> [!IMPORTANT]
+> Within OpenShift, Portworx shared volumes make use of the Container Storage Interface (CSI) via the `com.openstorage.pxd` provisioner. OpenShift 3.11 uses v0.3 of the CSI spec, which is a beta release, so use of shared volumes is not recommended within a production environment.
+
+## Creating a persistent volume claim
+
+The following example shows a persistent volume claim (pvc) spec using a shared storage class. Note the `ReadWriteMany` access mode:
+
+```none
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+   name: pvc-shared
+   annotations:
+     volume.beta.kubernetes.io/storage-class: portworx-repl1-shared
+spec:
+   accessModes:
+     - ReadWriteMany
+   resources:
+     requests:
+       storage: 10Gi
+```
+
+Once you've created the pvc, verify that its status shows as `Bound` to confirm it has been provisioned by Portworx:
+
+```none
+$ oc get pvc
+NAME             STATUS    VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS            AGE
+pvc-shared   Bound     pvc-3654b857-efa1-11e9-8422-fa163e52fd0e   10Gi       RWX            portworx-repl1-shared   1m
+```
+
```

Comment:
Should we be pointing out and showing the PV that gets created when the PVC dynamically provisions the volume?